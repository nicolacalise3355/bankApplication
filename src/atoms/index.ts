export * from "./CardBox"

export * from "./MessageBox"