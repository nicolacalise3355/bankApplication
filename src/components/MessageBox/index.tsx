import React, { useState } from 'react'
import { Alert, Modal, Pressable, Text, View } from 'react-native'
import { MessageBoxStyle } from './style'
import { MessageBoxProps } from '../../interfaces/AtomsInterfaces'
import { timestampToDate } from '../../services/Utility'

export const MessageBox = ({ timestamp, title, message, callbackClose }: MessageBoxProps) => {
  const [modalVisible, setModalVisible] = useState(false);

  return (
    <View style={MessageBoxStyle.container}>
      <Pressable onPress={callbackClose} testID='messageBox'>
        <View style={MessageBoxStyle.containerClose}>
          <Text style={MessageBoxStyle.close}>X</Text>
        </View>
      </Pressable>
      <Pressable onPress={() => setModalVisible(!modalVisible)}>
        <View style={MessageBoxStyle.containerText}>
          <Text style={MessageBoxStyle.simpleText}>Message: {title}</Text>
        </View>
      </Pressable>
      <Modal
        animationType="slide"
        transparent={false}
        visible={modalVisible}
        onRequestClose={() => {
          Alert.alert('Modal has been closed.');
          setModalVisible(!modalVisible);
        }}>
        <View style={MessageBoxStyle.modalContainer}>
            <View style={MessageBoxStyle.modalTextContainer}>
              <Text style={MessageBoxStyle.modalTitleText}>{title} : {timestampToDate(timestamp)}</Text>
              <Text style={MessageBoxStyle.modalText}>{message}</Text>
            </View> 
            <Pressable
              style={MessageBoxStyle.modalButtonClose}
              onPress={() => setModalVisible(!modalVisible)}>
              <Text>Close Modal</Text>
            </Pressable>
        </View>
      </Modal>
    </View>
  )
}
