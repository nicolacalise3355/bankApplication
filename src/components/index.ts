export * from "./TransactionsList"
export * from "./TransactionsList/TransactionsListLoader";