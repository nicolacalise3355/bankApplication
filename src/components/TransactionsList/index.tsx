import React from 'react'
import { View, Text, ScrollView } from 'react-native'
import { TransactionsListProps } from '../../interfaces/AtomsInterfaces'
import { TransactionBox } from '../../atoms'
import { TransactionsListStyle } from './style'

export const TransactionsList = ({ transactions }: TransactionsListProps) => {
  return (
    <ScrollView style={TransactionsListStyle.containerScroll}>
      <View style={TransactionsListStyle.container}>
        {
          transactions.map((t) => {
            return(
              <TransactionBox transaction={t} key={t.id}/>
            )
          })
        }
      </View>
    </ScrollView>
  )
}
