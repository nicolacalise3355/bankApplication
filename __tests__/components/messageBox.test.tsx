import renderer from 'react-test-renderer';
import { MessageBox } from '../../src/components';
import { render, fireEvent } from '@testing-library/react-native';
import React from 'react';

test('renders correctly', () => {
  const tree = renderer.create(<MessageBox  timestamp={'123456'} title={'Title'} message={'Message'} callbackClose={() => undefined} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('callback test', () => {
  const mockOnPress = jest.fn();

  const { getByTestId  } = render(<MessageBox timestamp={'123456'} title={'Title'} message={'Message'} callbackClose={mockOnPress} />);
  fireEvent.press(getByTestId('messageBox'));
  expect(mockOnPress).toHaveBeenCalled();
});

