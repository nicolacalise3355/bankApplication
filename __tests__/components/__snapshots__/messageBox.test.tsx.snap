// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    {
      "backgroundColor": "rgb(240, 240, 240)",
      "borderRadius": 16,
      "display": "flex",
      "gap": 2,
      "padding": 8,
      "paddingTop": 0,
      "width": "100%",
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    testID="messageBox"
  >
    <View
      style={
        {
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "flex-end",
          "padding": 4,
        }
      }
    >
      <Text
        style={
          {
            "fontSize": 18,
            "fontWeight": "700",
          }
        }
      >
        X
      </Text>
    </View>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
  >
    <View
      style={
        {
          "alignItems": "center",
          "display": "flex",
          "justifyContent": "center",
        }
      }
    >
      <Text
        style={
          {
            "fontSize": 18,
            "fontWeight": "600",
          }
        }
      >
        Message: 
        Title
      </Text>
    </View>
  </View>
</View>
`;
